public/static/app/perfdiscovery/PerfDiscoveryService.js

```diff
@@ -0,0 +1,207 @@
+mciModule.factory('PerfDiscoveryService', function($q, ApiV1, ApiTaskdata) {
+  var _respData = function(resp) {
+    return resp.data
+  }
+
+  // TODO Should be part of `VersionResource`
+  // :parm version: version object
+  // :returns: Array of {taskId, taskName, buildName}
+  function extractTasks(version) {
+    return _.reduce(version.builds, function(items, build) {
+      // Going over dict of tasks ({taskName: data})
+      return items.concat(_.map(build.tasks, function(taskData, taskName) {
+        return {
+          taskId: taskData.task_id,
+          taskName: taskName,
+          buildName: build.name,
+        }
+      }))
+    }, [])
+  }
+
+  // Makes series of HTTP calls and loads curent, baseline and history data
+  // :rtype: $q.promise
+  function queryData(tasks, version, baseline) {
+    var current = $q.all(_.map(tasks, function(task) {
+      return ApiTaskdata.getTaskById(task.taskId, 'perf')
+        .then(_respData, function() { return null })
+        .then(function(data) {
+          if (data) {
+            return $q.all({
+              ctx: $q.resolve(task),
+              current: data,
+              history: ApiTaskdata.getTaskHistory(task.taskId, 'perf')
+                .then(_respData, function(e) { return [] }),
+              baseline: ApiTaskdata.getTaskCommit({
+                projectId: data.project_id,
+                revision: baseline.revision,
+                variant: data.variant,
+                taskName: task.taskName,
+                name: 'perf',
+              }).then(_respData, function() { null }),
+            })
+          } else {
+            return null
+          }
+        })
+    }))
+
+    return $q.all({
+      tasks: current,
+    })
+  }
+
+  var seRe = new RegExp('[_ -](wt|wiredtiger|mmapv1|inmemory)$', 'i')
```

Comment:
Is the storage engine available as metadata in the source data at all? I'd like to avoid doing something like this in the client

Reply:
Me too. I copied it from 'Mathias view' py script, which uses the same API as I use. I didn't noticed `storageEngine` as metadata in these APIs.

Reply:
why don't we talk to david to figure out how to propagate this data more clearly.

Reply:
@tychoish , I'm not familiar with evergreen models, if you think we could extend build or task model (depending on perf team needs) with `storageEngine` attribute, it will be great. At this moment I ~have~ know no alternatives.

Reply:
We opened a ticket to get this metadata - could you add a comment on this line that it should be changed for EVG-2744?